content/en/blog/2022/apisix.md

```diff
@@ -0,0 +1,393 @@
+---
+
+This article introduces the Apache APISIX's `opentelemetry` plugin conceptand
+how to enable and deploy the plugin.
+
+## Background Information
+
+OpenTelemetry is an open source telemetry data acquisition and processing
+system. It not only provides various SDKs for application side telemetry data
+collection and reporting, but also provides data collection side for data
+receiving, processing and exporting. Export to any or more OpenTelemetry
+backends, such as Jaeger, Zipkin, and OpenCensus. You can view the list of
+plugins that have adapted the OpenTelemetry Collector in the
+[opentelemetry collector contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib)
+library.
+
+![Architecture-Present](/img/apisix/architecture-present.png)
+
+## Plugin Introduction
+
+The `opentelemetry` plugin of Apache APISIX implements Tracing data collection
+based on OpenTelemetry native standard (OTLP/HTTP), and sends it to
+OpenTelemetry Collector through HTTP protocol. Apache APISIX starts to support
+this feature in v2.13.0.
+
+One of OpenTelemetry's special features is that the Agent/SDK of OpenTelemetry
+is not locked with back-end implementation, which gives users flexibilities on
+choosing their own back-end services. In other words, users can choose the
+backend services they want, such as Zipkin and Jaeger, without affectting the
+application side.
+
+The `opentelemetry` plugin is located on the Agent side. It integrates the
+OpenTelemetry Agent/SDK and adopts its features in Apache APISIX. It can
+collect traced requests, generate `trace` and forward them to the OpenTelemetry
+Collector. It supports the `trace` protocol, and it will support the `logs` and
+`metrics` protocols of OpenTelemetry in the next version.
+
+## Enable the Plugin
+
+You need to enable `opentelemetry` plugin and modify collector configuration in
+`conf/config.yaml` configuration file.
+
+We assume that you have already deployed the OpenTelemetry Collector on the same
+node as the APISIX and enabled the [OTLP HTTP Receiver](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/README.md).
+
+> If you have not completed the deployment, you can refer to the Scenario
+> Example section in the next section to complete the deployment of
+> OpenTelemetry Collector.
+
+The default port of the OTLP HTTP Receiver is `4318`, and the address of the
+`collector` is the HTTP Receiver address of the OpenTelemetry Collector. For
+related fields, please refer to the [Apache APISIX official documentation](https://apisix.apache.org/docs/apisix/next/plugins/opentelemetry/).
+
+For specific configuration, please refer to the following example:
+
+```shell
+plugins
+  ... # Other plugins that have been enabled
+  - opentelemetry
+plugin_attr:
+  ...
+  opentelemetry:
+    trace_id_source: x-request-id
+    resource:
+      service.name: APISIX
+    collector:
+      address: 127.0.0.1:4318 # OTLP HTTP Receiver address
```

Comment:
out of curiosity, is default http or https? Is it configureable?

Reply:
It is default http now. By looking at the source code[1], I don't think https is supported. But if that is super desired feature, you can create an issue in the repo, and then we can add it to the next version.

[1]https://github.com/apache/apisix/blob/master/apisix/plugins/opentelemetry.lua#L71